include/deal.II/fe/mapping_manifold.h

```diff
@@ -0,0 +1,581 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2000 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__mapping_manifold_h
+#define dealii__mapping_manifold_h
+
+
+#include <deal.II/base/derivative_form.h>
+#include <deal.II/base/config.h>
+#include <deal.II/base/table.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/qprojector.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/fe/mapping.h>
+#include <deal.II/fe/fe_q.h>
+
+#include <cmath>
+
+DEAL_II_NAMESPACE_OPEN
+
+template <int,int> class MappingQ;
+
+
+/*!@addtogroup mapping */
+/*@{*/
+
+
+/**
+ * This class implements the functionality for Manifold conforming
+ * mappings. This Mapping computes the transformation between the
+ * reference and real cell by exploiting the geometrical information
+ * coming from the underlying Manifold object.
+ *
+ * Quadrature points computed using this mapping lie on the exact
+ * geometrical objects, and tangent and normal vectors computed using
+ * this class are tangent and normal to the underlying geometry. This
+ * is in constrast with the MappingQ class, which approximates the
+ * geometry using a polynomial of some order, and then computes the
+ * normals and tangents using the approximated surface.
+ *
+ * @author Luca Heltai, Wolfgang Bangerth, Alberto Sartori 2016
+ */
+template <int dim, int spacedim=dim>
+class MappingManifold : public Mapping<dim,spacedim>
+{
+public:
+  /**
+   * Constructor.
+   */
+  MappingManifold ();
+
+  /**
+   * Copy constructor.
+   */
+  MappingManifold (const MappingManifold<dim,spacedim> &mapping);
+
+  // for documentation, see the Mapping base class
+  virtual
+  Mapping<dim,spacedim> *clone () const;
+
+  /**
+   * Always returns @p true because the default implementation of functions in
+   * this class preserves vertex locations.
+   */
+  virtual
+  bool preserves_vertex_locations () const;
```

Comment:
update the documentation -- it preserves the locations because it assumes that vertices are already on the manifold.